### DISCUSSION TITLE
Why use the Publish/Subscribe pattern vs. event listeners?

### DISCUSSION THREAD

**Discussion by cfxd** [Shopify/dawn#2571] [General]
Hi all, I was just wondering why the Shopify team is effectively rolling their own solution to events when `dispatchEvent()` and `addEventListener()` are so widely used and supported?

**Comment by utkarshsaxena93** [comment]
Hey Cfxd,
Yep, we should have used `dispatchEvent` and `addEventListener` here. Thanks for nudging us in the correct direction.

**Comment by cfxd** [comment]
Bumping this. I'd be happy to write a PR if it will get attention!
